source/common/http/async_client_impl.cc

```diff
@@ -172,6 +174,8 @@ void AsyncStreamImpl::sendHeaders(RequestHeaderMap& headers, bool end_stream) {
 }
 
 void AsyncStreamImpl::sendData(Buffer::Instance& data, bool end_stream) {
+  streamInfo().addBytesSent(data.length());
```

Comment:
We're deliberately not accounting for headers?

Reply:
Per our offline discussion: We switch to codec-level byte meter. [wire_bytes](https://github.com/envoyproxy/envoy/blob/928b5b5a0ab4c04d62dc63b40900bc52ce23f73f/envoy/stream_info/stream_info.h#L447-L448) includes everything.